docs/coding-guidelines.md

```diff
@@ -95,6 +95,14 @@ To keep naming patterns consistent across the project, enumeration patterns are
   - `pcommon.ValueTypeStr` for `pcommon.ValueType`
   - `pmetric.MetricTypeGauge` for `pmetric.MetricType`
 
+## Optional fields
+
+When representing optional fields in the configuration, we distinguish two cases:
+1. For scalars (e.g. `int`, `string`, `bool`, `time.Duration`...), the zero value of the base type
+   is used to represent the absence of the field. The default value for this field may be different
+   from the zero value, so, when using the Go API you MUST follow the guidance on the "Default
+   Configuration" section below.
+2. For struct fields, you may use a pointer to struct to represent an optional sections.
```

Comment:
I'm wondering if we should suggest just using non-pointer values for now so we can be consistent across structs.

Our stable config modules don't use pointers for any struct fields, so this would allow us to be consistent. Once we've arrived at a decision, we can later indicate that pointer values (or optional types) should be used for types where zero values don't indicate absence of the field.